### PR TITLE
Milestone 4/category filter

### PR DESCRIPTION
**Pull request by MauricioRobayo**
This PR adds additional functionality to the application by allowing to filter books in the table by category:

1. Enhance the Redux store
    - [x] The Redux store should keep information about the current books filter. Name it `filter`. It can be a simple string with the name of the category. Remember to add the option `All` which will just remove the filter from the table.
    - [x] Create the `CHANGE_FILTER` action in `actions/index.js`
    - [x] Create a `filter.js` reducer which modifies `filter` in the Redux store. It should default to `All`.
2. Implement the `CategoryFilter` component
    - [x] The component should render a select box with filter categories.
    - [x] Remember to merge the book categories with the `All` option in the select box. `All` is not a valid book category and shouldn't be saved in the Redux store, although it should be a filter option.
3. Keep the filter value in the Redux store
    - [x] Render the `CategoryFilter` component in the `BooksList` component.
    - [x] Implement the `handleFilterChange` method which modifies the `filter` state in the Redux store. Pass it down as a prop to `CategoryFilter`.
    - [x] Implement a conditional method which decides which books should be shown in the table depending on the `filter` value from Redux store.

**Important**: I am working alone in this project since there was no one to pair me with, so I was approved to work on this final project alone.

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import categories from '../config'
```
Comment (shloch) [inline src/components/CategoryFilter.js:3]: so when you do like this, it picks up the `index.js` in the `config` folder ? 😊

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const Filter = ({ handleFilterChange }) => (
+  <select
+    id="filter"
+    onChange={event => handleFilterChange(event.target.value)}
+  >
+    {['All', ...categories].map(category => (
+      <option key={category} value={category}>
+        {category}
+      </option>
+    ))}
```
Comment (shloch) [inline src/components/CategoryFilter.js:14]: wise move

**Review by shloch — APPROVED** [review]
## Perfect execution @MauricioRobayo 🎅🎅
This is valid from me, 

### You can merge this PR 🎅🎅
Happy coding

File: src/components/CategoryFilter.js (hunk 1 above)
Comment (MauricioRobayo) [reply src/components/CategoryFilter.js:3]: That's right!

**Pull request merged**